components/tools/OmeroPy/test/integration/tablestest/test_tables.py

```diff
@@ -0,0 +1,334 @@
+
+        bcol = columns.BoolColumnI('boolcol', 'bool col')
+        bcol.values = [True, False]
+        dcol = columns.DoubleColumnI('doublecol', 'double col')
+        dcol.values = [0.25, 0.5]
+        lcol = columns.LongColumnI('longcol', 'long col')
+        lcol.values = [-1, -2]
+
+        scol = columns.StringColumnI('stringcol', 'string col', 3)
+        scol.values = ["abc", "de"]
+
+        larr = columns.LongArrayColumnI('longarr', 'longarr col', 2)
+        larr.values = [[-2, -1], [1, 2]]
+        farr = columns.FloatArrayColumnI('floatarr', 'floatarr col', 2)
+        farr.values = [[-0.25, -0.5], [0.125, 0.0625]]
+        darr = columns.DoubleArrayColumnI('doublearr', 'doublearr col', 2)
+        darr.values = [[-0.25, -0.5], [0.125, 0.0625]]
+
+        dscol = columns.DatasetColumnI('datasetcol', 'dataset col')
+        dscol.values = [110, 120]
+
+        mask = self.createMaskCol()
+
+        cols = [fcol, icol, rcol, wcol, pcol,
+                bcol, dcol, lcol, scol, mask,
+                larr, farr, darr, dscol]
+
+        table.initialize(cols)
+        table.setAllMetadata(metadata)
+
+        table.addData(cols)
+        data = table.readCoordinates([0, 1])
+
+        testf = data.columns[0].values
+        assert 10 == testf[0]
+        assert 20 == testf[1]
+        testi = data.columns[1].values
+        assert 30 == testi[0]
+        assert 40 == testi[1]
+        testr = data.columns[2].values
+        assert 50 == testr[0]
+        assert 60 == testr[1]
+        testw = data.columns[3].values
+        assert 70 == testw[0]
+        assert 80 == testw[1]
+        testp = data.columns[4].values
+        assert 90 == testp[0]
+        assert 100 == testp[1]
+
+        testb = data.columns[5].values
+        assert testb[0]
+        assert not testb[1]
+        testd = data.columns[6].values
+        assert 0.25 == testd[0]
+        assert 0.5 == testd[1]
+        testl = data.columns[7].values
+        assert -1 == testl[0]
+        assert -2 == testl[1]
+
+        tests = data.columns[8].values
+        assert "abc" == tests[0]
+        assert "de" == tests[1]
+
+        testm = data.columns[9]
+        self.checkMaskCol(testm)
+
+        testla = data.columns[10].values
+        assert [-2, -1] == testla[0]
+        assert [1, 2] == testla[1]
+        testfa = data.columns[11].values
+        assert [-0.25, -0.5] == testfa[0]
+        assert [0.125, 0.0625] == testfa[1]
+        testda = data.columns[12].values
+        assert [-0.25, -0.5] == testda[0]
+        assert [0.125, 0.0625] == testda[1]
+
+        testds = data.columns[13].values
+        assert 110 == testds[0]
+        assert 120 == testds[1]
+
+        ofile = table.getOriginalFile()
+        print("Created OriginalFile:", ofile.getId().val)
+
+        return table
+
+    def testAllColumnsAndMetadata(self):
+        """
+        Check the integrity of a created table created.
+        """
+
+        table = self.testCreateAllColumnsAndMetadata()
```

Comment:
Since `testCreateAllColumnsAndMetadata()` is being called here to create the table, it can probably be named `_testCreateAllColumnsAndMetadata()` so as not to be picked-up by pytest and run as it's own test (since then it gets run twice).